back/engines/commercial/smart_groups/spec/models/project_spec.rb

```diff
@@ -0,0 +1,23 @@
+require 'rails_helper'
+
+RSpec.describe Project, type: :model do
+  subject(:project) { create(:project) }
+
+  let(:user) { create(:user) }
+
+  describe 'before_destroy' do
+    it 'destroys any smart group that refers to this project' do
```

Comment:
So, this seems to answer part of [my question](https://github.com/CitizenLabDotCo/citizenlab/pull/1858/files#r856258054). We destroy any smart group that refers to a project to be destroyed.